apps/bot/src/giveaway/giveaway.ts

```diff
@@ -0,0 +1,276 @@
+import { Hashira } from "@hashira/core";
+import type { GiveawayParticipant, GiveawayReward } from "@hashira/db";
+import { type Duration, addSeconds } from "date-fns";
+import {
+  ActionRowBuilder,
+  ButtonBuilder,
+  ButtonStyle,
+  EmbedBuilder,
+  time,
+} from "discord.js";
+import { base } from "../base";
+import { durationToSeconds, parseDuration } from "../util/duration";
+import { ensureUserExists } from "../util/ensureUsersExist";
+import { waitForButtonClick } from "../util/singleUseButton";
+import {
+  endGiveaway,
+  giveawayButtonRow,
+  leaveButtonRow,
+  parseRewards,
+  updateGiveaway,
+} from "./util";
+
+export const giveaway = new Hashira({ name: "giveaway" })
+  .use(base)
+  .command("givek", (command) =>
+    command
+      .setDMPermission(false)
+      .setDescription("Tworzenie giveawayów z różnymi nagrodami")
+      .addString("nagrody", (rewards) =>
+        rewards.setDescription(
+          "Nagrody do rozdania, oddzielone przecinkami, gdy nagród jest więcej to piszemy np. 2x200 punktów",
+        ),
+      )
+      .addString("czas_trwania", (duration) =>
```

Comment:
Nitpick but we usually have dashes in command names/params so I'd change this to:
```suggestion
      .addString("czas-trwania", (duration) =>
```